app/src/androidTest/java/com/appunite/loudius/util/MockWebServerRule.kt

```diff
@@ -0,0 +1,93 @@
+package com.appunite.loudius.util
+
+import android.util.Log
+import com.appunite.loudius.di.TestInterceptor
+import io.mockk.MockKMatcherScope
+import io.mockk.every
+import io.mockk.mockk
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import okhttp3.mockwebserver.SocketPolicy
+import org.intellij.lang.annotations.Language
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import strikt.api.Assertion
+import strikt.api.expectThat
+import strikt.assertions.isNotNull
+
+private const val TAG = "MockWebServerRule"
+
+class MockWebServerRule : TestRule {
+
+    val dispatcher: Dispatcher = mockk {
+        every { shutdown() } returns Unit
+        every { peek() } returns MockResponse().apply { this.socketPolicy = SocketPolicy.KEEP_OPEN }
+        every { dispatch(any()) } answers {
+            val request = it.invocation.args[0] as RecordedRequest
+            Log.w(TAG, "Request is not mocked: ${request.method} ${request.path}")
+            MockResponse().setResponseCode(404)
+        }
+    }
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                MockWebServer().use { server ->
+                    server.dispatcher = dispatcher
+                    TestInterceptor.testInterceptor = UrlOverrideInterceptor(server.url("/"))
+                    Log.v(TAG, "TestInterceptor installed")
+                    try {
+                    } finally {
+                        base.evaluate()
+                        Log.v(TAG, "TestInterceptor uninstalled")
+                        TestInterceptor.testInterceptor = null
+                    }
```

Comment:
We cannot just put a finally block here. We want to ensure that `TestsInterceptor` has been uninstalled and that the `TestInterceptor.testInterceptor` field is set to null.
@jacek-marchwicki correct me if I'm wrong :D

Reply:
I don't think `finally` is executed if the exceptions occurs outside try/catch block. 
<img width="661" alt="Screenshot 2023-05-09 at 12 43 26" src="https://github.com/appunite/Loudius/assets/18102775/b5957439-9720-4b4d-9def-255ebbdfe529">
<img width="626" alt="Screenshot 2023-05-09 at 12 43 44" src="https://github.com/appunite/Loudius/assets/18102775/e164c74f-5d0f-4157-bbb3-c3fc641382d3">

Reply:
This should be:
```kotlin
try {
  base.evaluate()
} finally {
  Log.v(TAG, "TestInterceptor uninstalled")
  TestInterceptor.testInterceptor = null
}
```

instead of:
``kotlin
try {
} finally {
  base.evaluate()
  Log.v(TAG, "TestInterceptor uninstalled")
  TestInterceptor.testInterceptor = null
}
```